### ISSUE TITLE
Make strongbox port configurable from etc/conf/strongbox.xml

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#598]
This is configurable via the [`etc/conf/strongbox.xml`](https://github.com/strongbox/strongbox/blob/master/strongbox-resources/strongbox-storage-api-resources/src/main/resources/etc/conf/strongbox.xml#L8) file. However, at the moment, this configuration property is not taken into consideration when starting Strongbox via the [`strongbox-assembly/strongbox-distribution`](https://github.com/strongbox/strongbox-assembly/tree/master/strongbox-distribution). This should be controlled when Jetty is started viat the [JettyLauncher](https://github.com/strongbox/strongbox-assembly/blob/master/strongbox-jetty-booter/src/main/java/org/carlspring/strongbox/servers/jetty/JettyLauncher.java) class.

The best way to test the [`strongbox-assembly/strongbox-distribution`](https://github.com/strongbox/strongbox-assembly/tree/master/strongbox-distribution), is to run the [`build-and-start`](https://github.com/strongbox/strongbox-assembly/blob/master/build-and-start) script.

**Comment by fuss86** [comment]
Additionally, it'd be really cool to be able to configure `jetty-maven-plugin` in the `strongbox/strongbox-web-core` to use value provided in `etc/conf/strongbox.xml`.

**Comment by carlspring** [comment]
We should not proceed with this task before #755.

**Comment by carlspring** [comment]
@fuss86 / @adavid9  @ptirador : Is this now obsolete with the migration to Spring Boot and YAML-based configuration? Is it configurable now? If not, perhaps we should update the description...?

**Comment by fuss86** [comment]
@carlspring in my opinion we have proper property `server.port` in `application.yaml` but we should get rid of `configuration.port` from `strongbox.yaml`

**Comment by carlspring** [comment]
I agree. This should be a simple task now. This should be configurable through the REST API and UI.

@adavid9 : When implementing the Spring Boot support did you do any work related to this?

**Comment by adavid9** [comment]
@carlspring Hi, I am not sure about which issue you're talking about if you could remind me it, but I am totally agree with @fuss86 If we use the `server.port` in the `application.yaml` the `configuration.port` is useless.

**Comment by carlspring** [comment]
The configuration controller will need to be changed to use the correct property now and we should drop the old property. This might affect the `strongbox-web-integration-tests`, although it's highly unlikely. 

@adavid9 : Would you like to look into this?

**Comment by adavid9** [comment]
@carlspring Yes I can do it.

**Comment by carlspring** [comment]
Great! Thanks! :)

**Comment by fuss86** [comment]
@carlspring, currently Spring Boot uses `server.port` from `application.yaml` which is embedded in the `jar` file, you can't make it 

> configurable through the REST API and UI.

unless we will decide to use external `application.yaml` file or read the port from other external file where we are allowed to write. You can't write to the configuration file embedded in `jar` package.

**Comment by fuss86** [comment]
https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html

**Comment by adavid9** [comment]
@fuss86 What do you propose, should we leave it for now until we will decide to use the external `application.yaml` file?

**Comment by carlspring** [comment]
@strongbox/core-developers , @adavid9 , @ptirador ,

I think that we need to re-work things, so that:
* [ ] When we package the `strongbox-distribution`, it's in such a way that the `application.yaml` file ends up in the `etc/conf` directory inside the distribution
* [ ] The `StrongboxConfigurationController` (?) can modify this value.
* [ ] This is configurable via the UI.
* [ ] Clearly, changing this value through the UI will mean that the service will need to be restarted and when the user does this, they should be prompter whether, or not they'd like to restart now.

What do you guys think about this?

**Comment by steve-todorov** [comment]
This task will open up other issues like the environment variables need to be taken into account so that they take precedence over the configuration files. 

We should probably come back to this task later.
